Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -284,7 +310,7 @@ public void Release(IPooledConnection connection)
                 }
                 if (_connectionValidator.IsConnectionReusable(connection))
                 {
-                    if (IsPoolFull())
+                    if (IsIdlePoolFull())
```

Comment:
Checking IdlePoolSize upon Release calls would cause most of the connections to be destroyed with the default value of 10 given the MaxPoolSize is set to Infinite. I think that we shall either check against MaxPoolSize here or change the default value of 10.

Reply:
But`MaxPoolSize = infinite, idlePoolSize = 10` was not changed. This is what the old driver was using. Also the same behaviour for old users.

I wrote in the Config settings that it is highly suggested to not use poolSize = Infinite when using async API.

Reply:
Ok.